File: 238-product-of-array-except-self/product-of-array-except-self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int pro=1;
        int count=0;
        vector<int>ans(nums.size(),0);
        for(int i=0;i<nums.size();i++){
            if(nums[i]==0){
                count++;
            }
            if(nums[i]!=0){
            pro=pro*nums[i];
            }
            
        }
        if(count>1)return ans;
        if(count==1){
            for(int i=0;i<nums.size();i++){
                if(nums[i]==0){
                    ans[i]=pro;
                }
            }
        }
      else {
        for(int i=0;i<nums.size();i++){
            ans[i]=pro/nums[i];
        }
      }
      return ans;
    }
};

File: 238-product-of-array-except-self/product-of-array-except-self.py
class Solution(object):
    def productExceptSelf(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        n=len(nums)
        ans=[0]*n
        product=1
        count=0
        for x in nums:
           if x ==0:
               count=count+1

           else:
               product=product*x

        if count>1:
             return ans

        if count==1:
            for i in range(n):
                if nums[i]==0:
                   ans[i]=product
            return ans
 
        for i in range(n):
            ans[i]=product/nums[i]

        return ans



